src/Microsoft.PowerShell.Commands.Utility/commands/utility/NewGuidCommand.cs

```diff
@@ -9,16 +11,46 @@ namespace Microsoft.PowerShell.Commands
     /// <summary>
     /// The implementation of the "new-guid" cmdlet.
     /// </summary>
-    [Cmdlet(VerbsCommon.New, "Guid", HelpUri = "https://go.microsoft.com/fwlink/?LinkId=2097130")]
+    [Cmdlet(VerbsCommon.New, "Guid", DefaultParameterSetName = "Default", HelpUri = "https://go.microsoft.com/fwlink/?LinkId=2097130")]
     [OutputType(typeof(Guid))]
-    public class NewGuidCommand : Cmdlet
+    public class NewGuidCommand : PSCmdlet
     {
         /// <summary>
-        /// Returns a guid.
+        /// Generates an instance of the Guid structure whose value is all zeros.
+        /// </summary>
+        [Parameter(ParameterSetName = "Empty")]
+        public SwitchParameter Empty { get; set; }
+
+        /// <summary>
+        /// Converts a string to a Guid.
+        /// </summary>
+        [Parameter(Position = 0, ValueFromPipeline = true, ParameterSetName = "FromString")]
+        [System.Diagnostics.CodeAnalysis.AllowNull]
+        public string FromString { get; set; }
```

Comment:
```suggestion
        public string InputObject { get; set; }
```

I'd recommend changing the parameter name to match other cmdlets.

Reply:
done